base_geoengine/static/src/js/views/geoengine_view.js

```diff
@@ -788,18 +814,37 @@ var GeoengineView = View.extend(geoengine_common.GeoengineMixin, {
     },
 
     open_record: function (feature, options) {
+        var self = this;
         var attributes = feature.get('attributes');
+        var model = attributes.model;
         var oid = attributes.id;
-        if (this.dataset.select_id(oid)) {
-            this.do_switch_view('form', null, options); //, null, { mode: "edit" });
-        } else {
-            this.do_warn("Geoengine: could not find id#" + oid);
+
+        if (model === this.model) {
+            if (this.dataset.select_id(oid)) {
+                this.do_switch_view('form', null, options); //, null, { mode: "edit" });
+            } else {
+                this.do_warn("Geoengine: could not find id#" + oid);
+            }
+        } 
+        else {
+            var model_obj = new Model(model);
+            model_obj.call('get_formview_id', [[oid], self.dataset.context]).then(function(view_id){
+                var pop = new common.FormViewDialog(self, {
+                    res_model: model,
+                    res_id: oid,
+                    context: self.dataset.context,
+                    title: _t("Open: ") + attributes.name,
```

Comment:
@lmarion-source the `name` attribute is not present by default. As proposed above, you should use `display_name`.